docs/console/incremental-guide.md

```diff
@@ -0,0 +1,571 @@
+# Writing Console Apps: An Incremental Guide
+
+This guide will show you how to use the .NET CLI tooling to build cross-platform console apps incrementally.  It will start with the most basic console app and eventually span multiple projects, including testing.
+
+If you're unfamiliar with the .NET CLI toolset, read [the overview](overview.md).  It's also helpful to have an understanding of the [console app paradigm](paradigm.md) and how [native binary compilation](single-binaries.md) works.
+
+## Prerequisites
+
+Before you begin, ensure you have the [latest .NET CLI tooling](http://dotnet.github.io/getting-started/).  You'll also need a text editor.
+
+## Hello, Console App!
+
+First, navigate to or create a new folder with a name you like.  "incremental" is the name chosen for the sample code, which can be found [here](https://github.com/dotnet/core-docs/samples/core-projects/console-apps/incremental).
+
+Open up a command prompt and type the following:
+
+```
+$ dotnet new
+$ dotnet restore
+$ dotnet run
+```
+
+You should see the following:
+
+`Hello World!`
+
+Let's do a quick walkthrough of what happened.
+
+1. `$ dotnet init`
+
+   This created an up-to-date `project.json` file with NuGet dependencies necessary to build a console app.  It also created a `Program.cs`, a basic file containing the entry point for the application.
+   
+   `project.json`:
+   ```javascript
+   {
+        "version": "1.0.0-*",
+        "compilationOptions": {
+            "emitEntryPoint": true
+        },
+    
+        "dependencies": {
+            "Microsoft.NETCore.Runtime": "1.0.1-beta-*",
+            "System.IO": "4.0.11-beta-*",
+            "System.Console": "4.0.0-beta-*",
+            "System.Runtime": "4.0.21-beta-*"
+        },
+    
+        "frameworks": {
+            "dnxcore50": { }
+        }
+   }
+   ```
+   `Program.cs`:
+   ```csharp
+   using System;
+
+   namespace ConsoleApplication
+   {
+       public class Program
+       {
+           public static void Main(string[] args)
+           {
+               Console.WriteLine("Hello World!");
+           }
+       }
+   }
+   ```
+
+2. `$ dotnet restore`
+
+   This analyzed the `project.json` file, downloaded the dependencies stated in the file (or grabbed them from a cache on your machine), and wrote the `project.lock.json` file.  The `project.lock.json` file is necessary to be able to compile and run.
+   
+3. `$ dotnet run`
+
+   This JIT compiled the source, writing "Hello World!" to the console.
```

Comment:
We should only use JIT in its very specific common usage. dotnet run does a lot, so calling it 'JIT" will be confusing to folks.